packages/engine/lib/memory/src/ffi/mod.rs

```diff
@@ -1,20 +1,20 @@
-#![allow(clippy::missing_safety_doc)]
```

Comment:
Should the `ffi` module be inside the `arrow` module?

Reply:
`ffi` also directly uses `Memory`, so I would leave it as top-level module.

Reply:
Right but Arrow stuff directly uses Memory does it not?

If the FFI logic is strictly about Arrow then it should be in the arrow module. I don't think we should **just** organise based on dependencies

Reply:
`ffi` is exposing `CMemory`, which is defined as
```
#[repr(C)]
pub struct CMemory {
    pub ptr: *const u8,
    pub len: i64,
    pub memory: *const Memory,
}
```

`CMemory` itself is used for several functions in `ffi`. It's however possible to split `ffi` into `shared_memory::ffi` and `arrow::ffi`.